lib/buffer.js

```diff
@@ -20,6 +20,8 @@ function createPool() {
   poolOffset = 0;
 }
 
+// v8::String::kMaxLength defined in v8.h
+const kStringMaxLength = (1 << 28) - 16;
```

Comment:
Also set the value from `node_buffer.cc` using `v8::String::kMaxLength`. Sort of how `Buffer::kMaxLength` is now. This way the value is controlled by V8, and we don't have to worry about it.